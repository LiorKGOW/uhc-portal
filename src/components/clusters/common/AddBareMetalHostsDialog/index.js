import { connect } from 'react-redux';
import { AddBareMetalHostsDialog } from 'facet-lib';

import { closeModal } from '../../../common/Modal/ModalActions';
import shouldShowModal from '../../../common/Modal/ModalSelectors';
import withFeatureGate from '../../../features/with-feature-gate';
import { ASSISTED_INSTALLER_FEATURE } from '../../../../redux/constants/featureConstants';

const mapStateToProps = (state) => {
  const cluster = state.modal.data;
  return ({
    isOpen: shouldShowModal(state, 'add-bare-metal-hosts'),
    cluster,
  });
};

const mapDispatchToProps = dispatch => ({
  closeModal: () => dispatch(closeModal()),
});

export default withFeatureGate(
  connect(mapStateToProps, mapDispatchToProps)(AddBareMetalHostsDialog),
  ASSISTED_INSTALLER_FEATURE,
  () => null,
);
