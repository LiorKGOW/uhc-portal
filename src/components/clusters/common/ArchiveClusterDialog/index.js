import { connect } from 'react-redux';

import {
  clearClusterArchiveResponse,
  archiveCluster,
} from '../../../../redux/actions/clustersActions';
import ArchiveClusterDialog from './ArchiveClusterDialog';
import { closeModal } from '../../../common/Modal/ModalActions';

const mapStateToProps = (state) => {
  const modalData = state.modal.data;
  return ({
    archiveClusterResponse: state.clusters.archivedCluster,
    subscriptionID: modalData.subscriptionID ? modalData.subscriptionID : '',
    name: modalData.name ? modalData.name : '',
  });
};

const mapDispatchToProps = dispatch => ({
  submit: (subscriptionID, name) => {
    dispatch(archiveCluster(subscriptionID, name));
  },
  resetResponse: () => dispatch(clearClusterArchiveResponse()),
  closeModal: () => dispatch(closeModal()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ArchiveClusterDialog);
