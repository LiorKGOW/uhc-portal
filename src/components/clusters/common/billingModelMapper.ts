/* eslint-disable camelcase */
import { RelatedResource } from '~/types/accounts_mgmt.v1';
import { BillingModel } from '~/types/clusters_mgmt.v1';

export const clusterBillingModelToRelatedResource = (
  clusterBillingModel?: BillingModel | string,
): RelatedResource.billing_model | undefined =>
  !['', undefined].includes(clusterBillingModel)
    ? (clusterBillingModel as unknown as RelatedResource.billing_model)
    : undefined;
