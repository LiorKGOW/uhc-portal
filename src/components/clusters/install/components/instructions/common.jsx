import React from 'react';
import PropTypes from 'prop-types';
import {
  Alert,
} from 'patternfly-react';
import {
  Button,
  Popover,
} from '@patternfly/react-core';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import Download from '@axetroy/react-download';
import config from '../../../../../config';

const DownloadButton = ({ token }) => (
  <Button
    component="a"
    href={config.configData.installerURL}
    target="_blank"
    variant="primary"
    className="install--download-installer"
    disabled={!!token.error}
    tabIndex="-1"
  >
    Download Installer
  </Button>
);
DownloadButton.propTypes = {
  token: PropTypes.object.isRequired,
};

const TelemetryAlert = () => (
  <div className="alert alert-warning">
    <span className="pficon pficon-warning-triangle-o" />
    Red Hat collects a limited amount of telemetry data. By installing OpenShift Container
    Platform 4, you accept our data collection policy.
    {' '}
    <a href="https://github.com/openshift/telemeter/blob/master/docs/data-collection.md" target="_blank">
      Learn more
      {' '}
      <span className="fa fa-external-link" aria-hidden="true" />
    </a>
    {' '}
    about the data we collect.
  </div>
);

const TokenErrorAlert = ({ token }) => (
  <Alert
    variant="danger"
    className="install--errors"
    title={token.error.msg}
  >
    Please try again by refreshing the page.
    If the problem persists, please report the issue to
    {' '}
    <a href="mailto:***REMOVED***" target="_blank">
      ***REMOVED***
      {' '}
      <span className="fa fa-external-link" aria-hidden="true" />
    </a>
    .
  </Alert>
);
TokenErrorAlert.propTypes = {
  token: PropTypes.object.isRequired,
};

const GetStarted = ({ docURL }) => (
  <React.Fragment>
    <p>
      Follow the
      {' '}
      <a href={docURL} target="_blank">
        official documentation
        {' '}
        <span className="fa fa-external-link" aria-hidden="true" />
      </a>
      {' '}
      for detailed installation instructions.
    </p>
    <Button component="a" href={docURL} target="_blank" variant="primary">
      Get Started
    </Button>
    <p />
    <p>
      Relevant downloads are provided below.
    </p>
  </React.Fragment>
);
GetStarted.propTypes = {
  docURL: PropTypes.string.isRequired,
};

const CLISection = () => (
  <React.Fragment>
    <p>
      Download the OpenShift command-line tools and add them to your
      {' '}
      <code>PATH</code>
      .
    </p>
    <div>
      <a href={config.configData.commandLineToolsURL} target="_blank">
        <Button
          variant="secondary"
          className="install--download-cli"
          tabIndex="-1"
        >
          Download Command-Line Tools
        </Button>
      </a>
    </div>
    <p />
    <p>
      When the installer is complete you will see the console URL and credentials for
      accessing your new cluster. A
      {' '}
      <code>kubeconfig</code>
      {' '}
      file will also be generated for you to use with the
      {' '}
      <code>oc</code>
      {' '}
      CLI tools you downloaded.
    </p>
  </React.Fragment>
);

const PullSecretSection = ({
  tokenView, token, onCopy, copied,
}) => (
  <React.Fragment>
    <p>
      Download or copy your pull secret. The install program will prompt you for your pull
      secret during installation.
    </p>
    <div>
      <Download file="pull-secret" content={tokenView} style={{ display: 'inline' }}>
        <Button
          variant="secondary"
          disabled={!token || !!token.error}
        >
          Download Pull Secret
        </Button>
      </Download>
      <CopyToClipboard
        text={tokenView}
        onCopy={onCopy}
      >
        <span style={{ margin: '10px' }}>
          <button
            id="copyPullSecret"
            className="pf-c-button pf-m-link pf-m-inline install--copy-pull-secret"
            type="button"
            tabIndex="0"
            disabled={!token || !!token.error}
          >
            <span className="fa fa-paste" aria-hidden="true" />
            &nbsp;
            Copy Pull Secret
          </button>
          { copied && ' Copied!' }
        </span>
      </CopyToClipboard>
    </div>
  </React.Fragment>
);
PullSecretSection.propTypes = {
  tokenView: PropTypes.string.isRequired,
  token: PropTypes.object.isRequired,
  onCopy: PropTypes.func.isRequired,
  copied: PropTypes.bool.isRequired,
};

const RHCOSSection = ({ learnMoreURL, token }) => (
  <React.Fragment>
    <p>
      Download RHCOS to create machines for your cluster to use during installation.
      {' '}
      <a href={learnMoreURL} target="_blank">
        Learn more
        {' '}
        <span className="fa fa-external-link" aria-hidden="true" />
        .
      </a>
    </p>
    <p>
      <a href="https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.1/latest/" target="_blank">
        <Button
          variant="secondary"
          className="install--download-installer"
          disabled={!!token.error}
          tabIndex="-1"
        >
          Download RHCOS
        </Button>
      </a>
    </p>
  </React.Fragment>
);
RHCOSSection.propTypes = {
  learnMoreURL: PropTypes.string.isRequired,
  token: PropTypes.object.isRequired,
};

const WhatIsInstallerSection = ({ isIPI }) => (
  <React.Fragment>
    <Popover
      position="top"
      aria-label="What is the OpenShift installer?"
      bodyContent={(
        <div>
          The OpenShift Installer is a command-line installation wizard, prompting the user
          for values that it cannot determine on its own and providing reasonable defaults
          for everything else.
          { isIPI && (
            <React.Fragment>
              For more advanced users, the installer provides facilities
              for varying levels of customization.
              {' '}
              <a href="https://docs.openshift.com/container-platform/4.1/installing/installing_aws/installing-aws-customizations.html" target="_blank">
                Learn more
                {' '}
                <span className="fa fa-external-link" aria-hidden="true" />
              </a>
              .
            </React.Fragment>
          )}
        </div>
      )}
    >
      <a href="#" onClick={e => e.preventDefault()} className="popover-hover">
        <span className="pficon pficon-info" />
        {' '}
        What is the OpenShift installer?
      </a>
    </Popover>
  </React.Fragment>
);
WhatIsInstallerSection.propTypes = {
  isIPI: PropTypes.bool.isRequired,
};

export {
  CLISection,
  DownloadButton,
  GetStarted,
  PullSecretSection,
  RHCOSSection,
  TelemetryAlert,
  TokenErrorAlert,
  WhatIsInstallerSection,
};
