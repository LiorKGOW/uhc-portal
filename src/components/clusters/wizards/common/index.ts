export * from './constants';
export * from './utils';
export * from './Prerequisites/Prerequisites';
export * from './ValidationIconButton';
export * from './ValidationItem';
export { MachinePool as ClusterSettingsMachinePool } from './ClusterSettings/MachinePool/MachinePool';
export type { NodeLabel } from './ClusterSettings/MachinePool/NodeLabelsFieldArray';
export { ClusterUpdates } from './ClusterUpdates';
