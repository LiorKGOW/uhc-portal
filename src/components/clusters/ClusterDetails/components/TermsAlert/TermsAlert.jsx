import React, { Component } from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';
import { Alert, AlertActionLink } from '@patternfly/react-core';

import { subscriptionPlans, subscriptionStatuses } from '../../../../../common/subscriptionTypes';
import { buildUrlParams } from '../../../../../common/queryHelpers';

class TermsAlert extends Component {
  componentDidMount() {
    const { selfTermsReview } = this.props;
    if (this.isTermsReviewRquired()) {
      selfTermsReview();
    }
  }

  componentDidUpdate(prevProps) {
    const { subscription: prevSubscription } = prevProps;
    const { subscription, selfTermsReview } = this.props;
    if (get(prevSubscription, 'id') !== get(subscription, 'id') && this.isTermsReviewRquired()) {
      selfTermsReview();
    }
  }

  getTncAppURL = (baseURL) => {
    const currentHref = window.location.href;
    const params = {
      // redirect back to the same page
      redirect: currentHref,
      cancelRedirect: currentHref,
    };
    // baseURL contains params already.
    return `${baseURL}&${buildUrlParams(params)}`;
  };

  isTermsReviewRquired() {
    const { subscription } = this.props;
    const planID = get(subscription, 'plan.id');
    const status = get(subscription, 'status');

    return status !== subscriptionStatuses.DEPROVISIONED && (
      planID === subscriptionPlans.OSD
      || planID === subscriptionPlans.MOA
      || planID === subscriptionPlans.RHMI
    );
  }

  render() {
    const { selfTermsReviewResult } = this.props;

    if (this.isTermsReviewRquired()
      && selfTermsReviewResult.fulfilled
      && selfTermsReviewResult.terms_required) {
      const tncAppURL = this.getTncAppURL(selfTermsReviewResult.redirect_url);
      const redirectToTerms = (
        <AlertActionLink key="terms-alert-link" component="a" href={tncAppURL}>
          View Terms and Conditions
        </AlertActionLink>
      );

      return (
        <Alert
          id="terms-alert"
          variant="warning"
          isInline
          title="You must accept the Terms and Conditions"
          actionLinks={redirectToTerms}
        >
          <p>
            Access to this cluster will be read-only until you accept the Terms and Conditions.
          </p>
        </Alert>
      );
    }

    return null;
  }
}

TermsAlert.propTypes = {
  subscription: PropTypes.object.isRequired,
  selfTermsReview: PropTypes.func.isRequired,
  selfTermsReviewResult: PropTypes.object.isRequired,
};

export default TermsAlert;
