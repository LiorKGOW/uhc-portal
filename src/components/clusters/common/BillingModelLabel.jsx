import PropTypes from 'prop-types';

import get from 'lodash/get';

import { normalizedProducts, billingModels } from '../../../common/subscriptionTypes';

function BillingModelLabel({ cluster }) {
  const planId = get(cluster, 'subscription.plan.id');
  const billingModel = get(cluster, 'billing_model');
  const { ROSA, OSD, OSDTrial } = normalizedProducts;
  const { STANDARD, MARKETPLACE } = billingModels;
  const CCS = get(cluster, 'ccs.enabled');

  if (planId === ROSA) {
    return 'Through AWS';
  }

  if (planId === OSDTrial) {
    return 'Free trial, upgradeable';
  }

  // OSD non-ccs standard quota
  if (planId === OSD && billingModel === STANDARD) {
    return 'Subscription (yearly)';
  }

  // OSD CCS marketplace
  if (planId === OSD && billingModel === MARKETPLACE && CCS) {
    return 'Pay-as-you-go (hourly)';
  }

  return 'Standard';
}

BillingModelLabel.propTypes = {
  cluster: PropTypes.shape({
    product: PropTypes.shape({
      id: PropTypes.oneOf(Object.keys(normalizedProducts)).isRequired,
    }),
  }),
};

export default BillingModelLabel;
