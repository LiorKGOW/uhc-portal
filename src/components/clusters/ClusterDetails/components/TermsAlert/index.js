import { connect } from 'react-redux';

import { selfTermsReview } from '../../../../../redux/actions/userActions';
import TermsAlert from './TermsAlert';

const mapStateToProps = state => ({
  selfTermsReviewResult: state.userProfile.selfTermsReviewResult,
});

const mapDispatchToProps = {
  selfTermsReview,
};

export default connect(mapStateToProps, mapDispatchToProps)(TermsAlert);
