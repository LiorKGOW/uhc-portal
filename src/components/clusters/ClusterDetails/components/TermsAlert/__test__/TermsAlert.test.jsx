import React from 'react';
import { render, screen, checkAccessibility } from '~/testUtils';
import TermsAlert from '../TermsAlert';
import * as Fixtures from './TermsAlert.fixtures';
import { buildUrlParams } from '../../../../../../common/queryHelpers';
import { normalizedProducts } from '../../../../../../common/subscriptionTypes';

describe('<TermsAlert />', () => {
  describe('TermsAlert OCP', () => {
    const props = { ...Fixtures, subscription: { plan: { id: normalizedProducts.OCP } } };
    it('renders an empty object', () => {
      const { container } = render(<TermsAlert {...props} />);
      expect(container).toBeEmptyDOMElement();
    });
  });

  describe('TermsAlert has no required terms', () => {
    const props = { ...Fixtures, selfTermsReviewResult: { terms_required: false } };

    it('renders an empty object', () => {
      const { container } = render(<TermsAlert {...props} />);
      expect(container).toBeEmptyDOMElement();
    });
  });

  describe('TermsAlert has required terms', () => {
    const locationSave = window.location;
    const currentUrl = 'http://cloud.openshift.com/details/123';
    const locationTest = new URL(currentUrl);
    delete window.location;
    window.location = locationTest;

    const selfTermsReview = jest.fn();

    beforeEach(() => {
      selfTermsReview.mockClear();
    });

    afterAll(() => {
      window.location = locationSave;
    });

    it('is accessible', async () => {
      const { container } = render(<TermsAlert {...Fixtures} />);
      await checkAccessibility(container);
    });
    it('should contain link to terms app', () => {
      const params = {
        redirect: currentUrl,
        cancelRedirect: currentUrl,
      };
      const redirectUrl = `${Fixtures.tncUrl}&${buildUrlParams(params)}`;
      render(<TermsAlert {...Fixtures} />);

      expect(screen.getByRole('link', { name: 'View Terms and Conditions' })).toHaveAttribute(
        'href',
        redirectUrl,
      );
    });
  });
});
