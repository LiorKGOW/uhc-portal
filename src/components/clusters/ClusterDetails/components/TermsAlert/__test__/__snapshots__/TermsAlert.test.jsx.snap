// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TermsAlert /> TermsAlert has required terms should render 1`] = `
<Alert
  actionLinks={
    <AlertActionLink
      component="a"
      href="https://www.redhat.com/wapps/tnc/ackrequired?site=ocm&event=register&redirect=http%3A%2F%2Fcloud.openshift.com%2Fdetails%2F123&cancelRedirect=http%3A%2F%2Fcloud.openshift.com%2Fdetails%2F123"
    >
      View Terms and Conditions
    </AlertActionLink>
  }
  id="terms-alert"
  isInline={true}
  title="You must accept the Terms and Conditions"
  variant="warning"
>
  <p>
    Access to this cluster will be read-only until you accept the Terms and Conditions.
  </p>
</Alert>
`;
